### ISSUE TITLE
Feature request - Make it easily extensible by the user

### ISSUE THREAD

**Issue by amelio-vazquez-reina** [datasert/vscode-texty#1]
This is a very cool extension. I have been looking for something like it here: [Using Python or TypeScript to transform text that I select in VSCode](https://stackoverflow.com/questions/59060907/using-python-to-transform-text-that-i-select-in-the-editor).

From what I understand though, the extension currently only supports a fixed, pre-defined set of string transformations or commands. How difficult would it be to make it **extensible** for the **end user** ?

For example, let's say that I want to define my own string functions:
```
function add_space_around_text(selected_text) {
    return "  " + selected_text + "  ";
}
```
and I'd like to register a command ID of my choice with that function (via settings, i.e. without having to re-install the extension, that is)

IMHO - That could make the extension more extensible and even more powerful. 

What do you think?

**Comment by brsanthu** [comment]
I had been thinking something like this and also kind of macro that you could create by concatenating list of commands which will be executed in sequence.

Either way, thank you for posting. Will update soon.

**Comment by brsanthu** [comment]
I have got initial version working. Due to some limitations on the vscode api front, we are not able to do couple of things.

Here is how it works.

1. Add custom commands to settings.json as below.

```
...
    "texty": {
        "customCommands": {
            "transformText": {
                "title": "Length of selection",
                "script": "sel.length"
            }
        }
    }
...
```

You can add as many commands as possible. When you make changes commands, you need to restart the vscode so we can register those commands again.

Due to api limitation, you cannot search for command in the command palette since as of now we cannot register command dynamically with a title or category of command.

But you can still keybind them using command id. The command id for above custom command will be `texty.custom.transformText`.

In addition to that, texty also adds additional command `Execute Custom Command` which shows all defined custom commands as quick pick options (here it shows the title) and user can select one and texty will execute it.

The script can contain any javascript script. You can use following variables in the script.

sel => current selection
moment => moment object import
_ (underscore) => lodash import so you can use this to manipulate texts.

Hope this works for you.

**Comment by brsanthu** [comment]
Released as version 1.1.1

**Comment by amelio-vazquez-reina** [comment]
This is awesome. Thanks @brsanthu! I'll give this a try soon
